src/plugins/intel_cpu/src/serialize.cpp

```diff
@@ -55,9 +55,22 @@ void ModelDeserializer::operator>>(std::shared_ptr<ov::Model>& model) {
     std::string xmlString;
     ov::Tensor dataBlob;
 
+    // get file size before seek content
+    _istream.seekg(0, _istream.end);
+    const size_t file_size = _istream.tellg();
+    _istream.seekg(0, _istream.beg);
```

Comment:
The input `istream `also contains `CompiledBlobHeader`, we need skip it, please try below change:

```
diff --git a/src/plugins/intel_cpu/src/serialize.cpp b/src/plugins/intel_cpu/src/serialize.cpp
index a35a37378b..351cdce330 100644
--- a/src/plugins/intel_cpu/src/serialize.cpp
+++ b/src/plugins/intel_cpu/src/serialize.cpp
@@ -56,15 +56,16 @@ void ModelDeserializer::operator>>(std::shared_ptr<ov::Model>& model) {
     ov::Tensor dataBlob;
 
     // get file size before seek content
+    const size_t _pos = _istream.tellg();
     _istream.seekg(0, _istream.end);
     const size_t file_size = _istream.tellg();
-    _istream.seekg(0, _istream.beg);
+    _istream.seekg(_pos, _istream.beg);
 
     StreamSerialize::DataHeader hdr = {};
     _istream.read(reinterpret_cast<char*>(&hdr), sizeof hdr);
 
     // check if model header contains valid data
-    bool isValidModel = (hdr.custom_data_offset == sizeof(hdr)) &&
+    bool isValidModel = (hdr.custom_data_offset == sizeof(hdr) + _pos) &&
                         (hdr.custom_data_size == hdr.consts_offset - hdr.custom_data_offset) &&
                         (hdr.consts_size == hdr.model_offset - hdr.consts_offset) &&
                         (hdr.model_size = file_size - hdr.model_offset);
```